src/thinblock.cpp

```diff
@@ -1101,6 +1101,36 @@ void SendXThinBlock(CBlock &block, CNode *pfrom, const CInv &inv)
     pfrom->blocksSent += 1;
 }
 
+bool IsThinBlockValid(const CNode* pfrom, const std::vector<CTransaction>& vMissingTx, const CBlockHeader& header)
+{
+    // Check that that there is at least one txn in the xthin and that the first txn is the coinbase
+    if(vMissingTx.size() <= 0)
+    {
+        LogPrintf("No Transactions found in thinblock or xthinblock %s from peer %s (id=%d).\n",
+            header.GetHash().ToString(), pfrom->addrName.c_str(), pfrom->id);
+        return false;
+    }
+    if(!vMissingTx[0].IsCoinBase())
+    {
+        LogPrintf("First txn is not coinbase for thinblock or xthinblock %s from peer %s (id=%d).\n",
+            header.GetHash().ToString(), pfrom->addrName.c_str(), pfrom->id);
+        return false;
+    }
+
+    // check block header
+    CValidationState state;
```

Comment:
Using a state here makes me wonder why the other errors above were done using LogPrintf instead of using a state too.
Since 100% of the usages end up doing an `error()` call on failure, it would likely be easier to pinpoint the issue if you just used the CValidationState to throw the issue up. That also means you can avoid passing in the CNode.